tools/sense_studio/testing.py

```diff
@@ -0,0 +1,133 @@
+    # If classifier checkpoint exist, get the path of sub-directory from checkpoints
+    classifiers = [os.path.join('checkpoints', os.path.basename(d)) for d in glob.glob(f"{path}/checkpoints/*")
+                   if os.path.exists(os.path.join(d, 'best_classifier.checkpoint'))
+                   and os.path.isdir(d)]
+    classifiers.extend(['checkpoints/'] if os.path.exists(default_checkpoint) else [])
+    classifiers = natsorted(classifiers, alg=ns.IC)
+
+    return render_template('testing.html', project=project, path=path, output_path_prefix=output_path_prefix,
+                           classifiers=classifiers)
+
+
+@testing_bp.route('/start-testing', methods=['POST'])
+def start_testing():
+    data = request.json
+    path_in = data['inputVideoPath'] or ''
+    path_out = data['outputVideoName'] or ''
+    title = data['title']
+    path = data['path']
+    custom_classifier = os.path.join(path, data['classifier'])
+
+    config = project_utils.load_project_config(path)
+    output_dir = os.path.join(path, 'output_videos')
+    os.makedirs(output_dir, exist_ok=True)
+
+    if path_out:
+        path_out = os.path.join(output_dir, path_out + '.mp4')
+
+    ctx = multiprocessing.get_context('spawn')
+
+    global queue_testing_output
+    global queue_signal
+    queue_testing_output = ctx.Queue()
+    queue_signal = ctx.Queue()
+
+    testing_kwargs = {
+        'path_in': path_in,
+        'path_out': path_out,
+        'custom_classifier': custom_classifier,
+        'title': title,
+        'use_gpu': config['use_gpu'],
+        'video_frames': queue_testing_output.put,
+        'signal_queue': queue_signal,
+    }
+
+    global test_process
+    test_process = ctx.Process(target=run_custom_classifier, kwargs=testing_kwargs)
+    test_process.start()
+
+    return jsonify(success=True)
+
+
+@testing_bp.route('/cancel-testing')
+def cancel_testing():
+    global test_process
+    global queue_signal
+    if test_process:
+        # Send signal to stop inference
+        queue_signal.put(-1)
+
+        # Wait until process is complete
+        test_process.join()
+        test_process.terminate()
+        test_process = None
+        queue_signal.close()
+
+    return jsonify(success=True)
+
+
+@socketio.on('stream_video', namespace='/stream-video')
+def stream_video(msg):
+    global test_process
+    global queue_testing_output
+    try:
+        while test_process.is_alive():
+            try:
+                output = queue_testing_output.get(timeout=1)
+                if type(output) == str:
+                    emit('testing_logs', {'log': output})
+                else:
+                    # Encode frame as jpeg
+                    frame = cv2.imencode('.jpg', output)[1].tobytes()
+                    # Encode frame in base64 version and remove utf-8 encoding
+                    frame = base64.b64encode(frame).decode('utf-8')
+                    emit('testing_images', {'image': f"data:image/jpeg;base64,{frame}"})
```

Comment:
On the training page I did something similar, but on the javascript side. I would say we should treat both cases similar, but I don't mind which version we use.